packages/headless/src/features/facets/facet-search-set/facet-search-reducer-helpers.ts

```diff
@@ -143,6 +143,99 @@ export function handleCommerceFacetSearchFulfilled<
   }
 }
 
+export function handleCommerceFacetFieldSuggestionsFulfilled<
+  T extends FacetSearchResponse,
+>(
+  state: FacetSearchSetState<T>,
+  payload: {
+    facetId: string;
+    response: CommerceAPIResponse<T>;
+  },
+  requestId: string,
+  buildEmptyResponse: () => T
+) {
+  const {facetId, response} = payload;
+  let search = state[facetId];
+
+  if (!search) {
+    handleFacetSearchRegistration(state, {facetId}, buildEmptyResponse);
+    search = state[facetId];
+  } else if (search.requestId !== requestId) {
+    return;
+  }
+
+  search.isLoading = false;
+  if ('success' in response) {
+    search.response = response.success;
+  }
+}
+
+export function handleCommerceFetchQuerySuggestionsFulfilledForRegularFacet<
+  T extends FacetSearchResponse,
+>(
+  state: FacetSearchSetState<T>,
+  payload: {
+    fieldSuggestionsFacets: {facetId: string; type: string}[];
+    query: string | undefined;
+  },
+  requestId: string,
+  buildEmptyResponse: () => T
+) {
+  if (!payload.fieldSuggestionsFacets) {
+    return;
+  }
+
+  payload.fieldSuggestionsFacets
+    .filter((fieldSuggestionFacet) => !(fieldSuggestionFacet.facetId in state))
+    .filter((fieldSuggestionFacet) => fieldSuggestionFacet.type === 'regular')
+    .forEach((fieldSuggestionFacet) => {
```

Comment:
yike, that's 3 iterations in the array here.
use a single for..of with continue statement instead

Reply:
Done in https://github.com/coveo/ui-kit/pull/3939/commits/5c3d8fd53ed4b79eaf25f9692ac6c62b802242f2